pkg/reconciler/route/reconcile_resources.go

```diff
@@ -134,6 +125,12 @@ func (c *Reconciler) reconcilePlaceholderServices(ctx context.Context, route *v1
 		createdServiceNames.Insert(desiredService.Name)
 	}
 
+	existingServices, err := c.getServices(route)
```

Comment:
Will this end up getting the service we've just created?

Reply:
ah but we are only deleting the diff so it shouldn't matter

Reply:
Yes, thanks for reminding me why I copied it back in an interim version of this :joy: 

/hold